docs/guides/error-handling.mdx

```diff
@@ -65,77 +69,255 @@ RESTATE_WORKER__INVOKER__RETRY_POLICY__INTERVAL=100ms \
+    Have a look at our [sagas guide](/guides/sagas) to learn more.
+</Admonition>
+
+
+## Cancellations are Terminal Errors
+
+You can cancel invocations via the [CLI](/operate/invocation#cancelling-invocations), UI and programmatically.
+When you cancel an invocation, it throws a terminal error in the handler processing the invocation the next time it awaits a Promise or Future of a Restate Context action (e.g. run block, RPC, sleep,...; `CombineablePromise` in TypeScript, `Awaitable` in Java).
+Unless caught, This terminal error will propagate up the call stack until it reaches the original caller.
+
+Here again, the handler needs to have [compensation logic](/guides/sagas) in place to make sure the system remains in a consistent state, when you cancel an invocation.
 
 ## Timeouts to bound response times
 
+Restate also has timeouts to bound the response time of a service invocation.
+
+There are two types of timeouts in Restate:
+
 ### Inactivity timeout
-Default 1 minute
 
+When the Restate Server does not receive a next journal entry from a running handler within the inactivity timeout, it will ask the handler to suspend.
 This timer guards against stalled service/handler invocations. Once it expires, Restate triggers a graceful termination by asking the service invocation to suspend (which preserves intermediate progress).
 
-The 'abort timeout' is used to abort the invocation, in case it doesn't react to the request to suspend.
-
-Can be configured using the humantime format.
+By default, the inactivity timeout is set to one minute.
 
 ### Abort timeout
-Default 1 minute
 
-This timer guards against stalled service/handler invocations that are supposed to terminate. The abort timeout is started after the 'inactivity timeout' has expired and the service/handler invocation has been asked to gracefully terminate. Once the timer expires, it will abort the service/handler invocation.
+This timer guards against stalled service/handler invocations that are supposed to terminate.
+The abort timeout is started after the 'inactivity timeout' has expired and the service/handler invocation has been asked to gracefully terminate.
+Once the timer expires, it will abort the service/handler invocation.
 
-This timer potentially interrupts user code. If the user code needs longer to gracefully terminate, then this value needs to be set accordingly.
+By default, the abort timeout is set to one minute.
+This timer potentially interrupts user code.
+If the user code needs longer to gracefully terminate, then this value needs to be set accordingly.
 
-Can be configured using the humantime format.
+<Admonition type="note" title="Long-running run blocks">
+    If you have long-running `ctx.run` blocks, you need to increase both timeouts to prevent the handler from terminating prematurely.
+</Admonition>
 
 
-## Cancellations are Terminal Errors
-If you handle Terminal Errors, you automatically handle cancellation signals
+### Configuring the timeouts
```

Comment:
You can set those on a services basis too with `restate svc config edit <SVC_NAME>`